Yoast/wordpress-seo
# Create classic editor data object and get more replacevars from wpseoReplaceVarsL10n

## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* We've implemented a data class for the classic editor, similar to the Gutenberg data class. A nice side effect is that this means we were able to get rid of a lot of ternaries in the PostDataCollector.
* Both in Gutenberg and the classic editor, the data from wpseoReplaceVarsL10n is used to fill a large part of the replacement variables in the store.
* We created a new initializeData function to make testing easier. This function returns an instance of the correct data class (classic or Gutenberg). 
* We've created a helper function (sendReplaceVarsToStore) that takes the data and sends only the data needed for the replace vars to the store (not slug and content).

## Test instructions

This PR can be tested by following these steps:

Use `define( 'YOAST_FEATURE_SNIPPET_PREVIEW', true );` in `wp-config.php` to see the new snippet editor.
In both the classic editor and Gutenberg, test the following:
* In the redux-logger, see that the replacevar field is filled with an object containing all possible replacevars, however not all values are present (some are empty strings).
* Verify that the replacement works in the snippet preview editor (for the filled variables).
* Put %%title%% and %%excerpt%% in the snippet preview editor, and then change the relevant elements in the post editor. Verify that the values in the snippet preview are updated.
* **Important**: Verify that the SEO analysis and readability analysis are still working as expected, especially when the content and title etc. are changed.
* Also open categories, pages and tags, and verify that there are no errors and things work as expected.


## Quality assurance


Part of the fix for #9531
